src/connectors/hierarchical-menu/connectHierarchicalMenu.js

```diff
@@ -149,22 +158,13 @@ export default function connectHierarchicalMenu(renderFn, unmountFn) {
           ],
         };
 
-        if (limit !== undefined) {
-          const currentMaxValuesPerFacet =
-            currentConfiguration.maxValuesPerFacet || 0;
-          if (showMoreLimit === undefined) {
-            widgetConfiguration.maxValuesPerFacet = Math.max(
-              currentMaxValuesPerFacet,
-              limit
-            );
-          } else {
-            widgetConfiguration.maxValuesPerFacet = Math.max(
-              currentMaxValuesPerFacet,
-              limit,
-              showMoreLimit
-            );
-          }
-        }
+        const currentMaxValuesPerFacet =
+          currentConfiguration.maxValuesPerFacet || 0;
+
+        widgetConfiguration.maxValuesPerFacet = Math.max(
+          currentMaxValuesPerFacet,
+          showMoreLimit
```

Comment:
We always use the max number for the limit. It means that we always over fetch even if the user don't set `showMore`. We could provide a more accurate value based on the `showMore` option.

```
showMore ? showMoreLimit : limit
```

Reply:
Good point!